packages/gotrue/lib/src/types/auth_exception.dart

```diff
@@ -24,3 +24,33 @@ class AuthException implements Exception {
 class AuthPKCEGrantCodeExchangeError extends AuthException {
   AuthPKCEGrantCodeExchangeError(String message) : super(message);
 }
+
+class AuthSessionMissingError extends AuthException {
```

Comment:
I would rename all these classes to Exceptions. From the documentations: [Exception](https://api.dart.dev/stable/3.3.3/dart-core/Exception-class.html) and [Error](https://api.dart.dev/stable/3.3.3/dart-core/Error-class.html). Exceptions should be caught, which is the case here. Errors are more for wrong code. Additionally, they implement `Exception` and not `Error`.